packwiz: init

Infinidoge opened the pull request:
###### Description of changes

Packages [packwiz](https://github.com/packwiz/packwiz), a tool for creating modpacks for Minecraft.

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

Infinidoge committed:
packwiz: init

Infinidoge commented:
First time adding a new package to the top-level, so let me know if I need to move around the attribute.

Infinidoge committed:
maintainers: add infinidoge

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,20 @@
+{ lib
+, buildGo118Module
+, fetchFromGitHub
+}:
+buildGo118Module rec {
```
```suggestion
}:

buildGo118Module rec {
```

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,20 @@
+}:
+buildGo118Module rec {
+  pname = "packwiz";
+  version = "unstable-2022-5-25";
```
```suggestion
  version = "unstable-2022-5-25";
  
```

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,20 @@
+    repo = "packwiz";
+    rev = "e71b63ea98283c8c1f0e03ee51ae40f452f22a61";
+    sha256 = "sha256-XwGacEVfQAduDCSMQFRw7Xnx4bND2zaV7l27B+2u5xg=";
+  };
```
```suggestion
  };
  
```

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,20 @@
+  };
+  vendorSha256 = "sha256-M9u7N4IrL0B4pPRQwQG5TlMaGT++w3ZKHZ0RdxEHPKk=";
+
+  meta = with lib; {
```
Missing homepage and description

legendofmiracles reviewed (changes requested):
Formatting needed more space and missing attributes.

But otherwise looks good!

legendofmiracles commented:
Result of `nixpkgs-review pr 174591` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>packwiz</li>
  </ul>
</details>

Infinidoge commented:
Requested changes have been added and squashed, and I'll look into shell completion in a moment.

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,25 @@
+  vendorSha256 = "sha256-M9u7N4IrL0B4pPRQwQG5TlMaGT++w3ZKHZ0RdxEHPKk=";
+
+  meta = with lib; {
+    description = "A command line tool for editing and distributing Minecraft modpacks, using a git-friendly TOML format.";
```
```suggestion
    description = "A command line tool for editing and distributing Minecraft modpacks, using a git-friendly TOML format";
```
[Small nitpick](https://github.com/NixOS/nixpkgs/blob/c46c3ccefa2013392ee41fd846ff7e03658bbc2b/CONTRIBUTING.md?plain=1#L43)

Infinidoge commented:
Fixed the description nitpick. Also went ahead and added the shell completions based on the package you linked. If it isn't too much trouble, could you quickly test them? I could see the proper completion scripts under `$out/share`, but I suspect my setup is being weird and not letting completions run properly.

legendofmiracles commented:
![image](https://user-images.githubusercontent.com/30902201/170405956-65764cd3-6737-42ca-95de-79f69b09c398.png)
Sure, works for me! You have to install the package, just building it will not install the completions

legendofmiracles commented on pkgs/tools/games/minecraft/packwiz/default.nix:
```diff
@@ -0,0 +1,37 @@
+    description = "A command line tool for editing and distributing Minecraft modpacks, using a git-friendly TOML format";
+    homepage = "https://packwiz.infra.link/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ infinidoge ];
```
```suggestion
    maintainers = with maintainers; [ infinidoge ];
    mainProgram = "packwiz";
```
Because of the unstable in the version which messes with `nix run`

Infinidoge replied:
Weird, it runs just fine with `nix run` from my testing. Isn't it supposed to grab from `pname`?

Infinidoge replied:
(Tested with `nix run github:Infinidoge/nixpkgs/new/packwiz#packwiz`, for what it's worth)

Infinidoge replied:
Though I can't reproduce, added since it is of no cost.

legendofmiracles replied:
You're right.. in my 5 months of inactivity something must have changed... Oh well

legendofmiracles commented on pkgs/top-level/all-packages.nix:
```diff
@@ -1356,6 +1356,8 @@ with pkgs;
     python = python3;
   };
 
+  packwiz = callPackage ../tools/games/minecraft/packwiz { };
```
So that when buildGoModule changes it's default to 18, it's easier to update

Sorry, I only saw this change in another PR after the first batch of changes

Infinidoge replied:
Weird, it runs just fine with `nix run` from my testing.

Infinidoge replied:
Whoops, wrong change.

Infinidoge commented:
Ah, I was just sourcing the file manually with the command in a Nix shell, that makes sense. Anything else need to be done?

legendofmiracles reviewed (approved):
Yes, sorry. Last things, after that I will merge.

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/nix-minecraft-making-minecraft-more-declarative/19335/1
